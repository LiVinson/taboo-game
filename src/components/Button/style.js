import styled from "styled-components"

/*
  3 types of buttons:

  button
  button primary (extension of button)
  button-lg (extension of button) or button-lg primary (tbd)

*/

export const StyledButton = styled.button`
  border: none;
  color: ${(props) => props.theme.color.secondary};
  background-color: ${(props) => props.theme.color.offWhite};
  width: 47%;
  padding: 0.6rem 0rem;
  border-radius: 2px;
  font-size: 2.3rem;
  font-family: var(--font-family--display);
  text-transform: uppercase;

  &:active {
    outline: none;
    filter: brightness(0.8);
    transform: translateY(2px);
  }
`
export const StyledPrimaryButton = styled(StyledButton)`
  color: ${(props) => props.theme.color.offWhite};
  background-color: ${(props) => props.theme.color.secondary};
  width: 47%;
  &:active {
    filter: brightness(1.2);
  }
`
