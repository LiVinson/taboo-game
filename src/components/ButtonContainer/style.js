import styled from "styled-components"


export const StyledButtonContainer = styled.div`
  margin-top: 1.5rem;
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
`