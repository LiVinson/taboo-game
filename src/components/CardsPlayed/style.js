import styled from "styled-components"

export const NoCardMessage = styled.p`
	font-size:2rem;
	color: ${props => props.theme.color.grayDark2};
`