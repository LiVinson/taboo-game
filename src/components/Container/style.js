import styled from 'styled-components'

export const StyledContainer = styled.div`
	position: relative;
	width: 100vw;

	min-height: 100vh;
	overflow: hidden;
	/* 
    Used for background image. Fills entire Container div. Before used so opacity does not impact children elements */
	&:before {
		display: block;
		content: '';
		height: 100%;
		width: 100%;
		background-image: url(/img/pattern-light.png);
		opacity: 0.6;
		position: absolute;
		top: 0;
		left: 0;
	}
`
