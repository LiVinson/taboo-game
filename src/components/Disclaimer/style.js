import styled from "styled-components"

export const StyledDisclaimer = styled.p`
    margin-top: 2.5rem;
    margin-bottom: .5rem;
    font-size: 1.4rem;
    color: ${props => props.theme.color.grayDark2};
    text-align: center;
    padding: 0 .5rem;
`