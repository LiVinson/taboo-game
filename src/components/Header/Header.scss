.header {
    background: none;
    font-family: 'Oswald', sans-serif;
    font-size: 3rem;
    color: var(--color-primary);
    text-shadow: 0px 1rem 1rem rgba(0,0,0,.6);
    text-transform:uppercase;
}

//For main homescreen only, display larger text.
.header--large {
    font-size: 6rem;
    text-align: center;
    margin: 1rem 0;
}



.subheading {
    font-size: 2.2rem;
    text-align: center;
    width: 30rem;
    margin: 0 auto;
    color: var( --color-gray-dark-1);

    font-weight: 400;
    font-family: 'Oswald', sans-serif;

}

.subheading__focus-text {
    color: var(--color-primary-light);
}
