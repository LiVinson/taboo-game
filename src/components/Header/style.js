import styled, { css } from 'styled-components'

export const StyledHeader = styled.header`
	margin-top: ${(props) => (props.large ? '3rem' : '2rem')};
	margin-bottom: ${(props) => (props.subheading ? '5rem' : '3rem')};
`

export const Title = styled.h1`
	background: none;
	font-family: ${(props) => props.theme.text.display};
	font-size: ${(props) => props.theme.fontSize.medium};
	color: ${(props) => props.theme.color.primary};
	text-transform: uppercase;
  font-weight: 300;

	${(props) =>
		props.large &&
		css`
			font-size: ${(props) => props.theme.fontSize.xlarge};
			text-align: center;
			margin: 1rem 0;
		`}
`
Title.displayName = 'Title'

export const Subheading = styled.h3`
  font-size: 2.4rem;
  text-align: center;
  width: 30rem;
  margin: 0 auto;
  font-weight: 300;
  color: ${(props) => props.theme.color.grayDark1};
  background-color: rgba(247, 249, 249, .5);
  line-height:1.3;
  
`
Subheading.displayName = 'Subbeading'

export const FocusSpan = styled.span`
	color: ${(props) => props.theme.color.primary};

`
FocusSpan.displayName = 'FocusSpan'
