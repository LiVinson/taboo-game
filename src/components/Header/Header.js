import React from "react"
import PropTypes from "prop-types"
import {  StyledHeader, Title, Subheading, FocusSpan} from "./style.js"


export default function Header({ location }) {
  //size of header and whether subheading included varies based on route location
  const homeOrEndRoute = location.pathname.includes("home") || location.pathname.includes("end")
  const homeRouteExact = location.pathname === ("/home")

  return (
    <StyledHeader large={homeOrEndRoute} subheading={homeRouteExact}>
      <Title large={homeOrEndRoute}>Taboo!</Title>
      {homeRouteExact && (
        <Subheading>
          The team game that’s all about what you{" "}
          <FocusSpan>say,</FocusSpan> and what you{" "}
          <FocusSpan>don’t!</FocusSpan>
        </Subheading>
      )}
    </StyledHeader>
  )
}

Header.propTypes = {
  location: PropTypes.object.isRequired,
}
