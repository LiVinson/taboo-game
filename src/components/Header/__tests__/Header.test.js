import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { ThemeProvider } from 'styled-components'

import theme from '../../../global-design/theme'
import Header from '../Header'
import { Title, Subheading } from 'components/Header/style.js'

//--------------Header--------------------
test('Header renders correctly on /home route', () => {
	const props = {
		location: {
			pathname: '/home',
		},
	}
	//Requires TP due to nested components accessing theme prop
	const wrapper = renderer
		.create(
			<ThemeProvider theme={theme}>
				<Header theme={theme} {...props} />
			</ThemeProvider>
		)
		.toJSON()

	expect(wrapper).toMatchSnapshot()
})

test('Header renders correctly on nested home route', () => {
	const props = {
		location: {
			pathname: '/home/create',
		},
	}
	const wrapper = renderer
		.create(
			<ThemeProvider theme={theme}>
				<Header theme={theme} {...props} />
			</ThemeProvider>
		)
		.toJSON()

	expect(wrapper).toMatchSnapshot()
})

test('Header renders correctly when not on home route', () => {
	const props = {
		location: {
			pathname: '/play/90210/12345',
		},
	}
	const wrapper = renderer
		.create(
			<ThemeProvider theme={theme}>
				<Header theme={theme} {...props} />
			</ThemeProvider>
		)
		.toJSON()

	expect(wrapper).toMatchSnapshot()
})

//Title
test('Title renders the larger centered title when large prop is true', () => {
	const props = {
		large: true,
	}
	const wrapper = shallow(<Title {...props} theme={theme} />)
	expect(wrapper).toHaveStyleRule('font-size', '7rem')
	expect(wrapper).toHaveStyleRule('text-align', 'center')
})

test('Title renders the smaller title when large is small', () => {
	const props = {
		large: false,
	}
	const wrapper = shallow(<Title {...props} theme={theme} />)
	expect(wrapper).toHaveStyleRule('font-size', '2.7rem')
})

//-------------- Subheading --------------------
test('Subheading renders correctly', () => {
	const wrapper = renderer.create(<Subheading theme={theme} />).toJSON()
	expect(wrapper).toMatchSnapshot()
})
