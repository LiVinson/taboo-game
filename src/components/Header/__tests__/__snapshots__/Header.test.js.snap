// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly on /home route 1`] = `
.c0 {
  margin-top: 3rem;
  margin-bottom: 5rem;
}

.c1 {
  background: none;
  font-family: 'Oswald',sans-serif;
  font-size: 2.7rem;
  color: #1982C4;
  text-transform: uppercase;
  font-weight: 300;
  font-size: 7rem;
  text-align: center;
  margin: 1rem 0;
}

.c2 {
  font-size: 2.4rem;
  text-align: center;
  width: 30rem;
  margin: 0 auto;
  font-weight: 300;
  color: #757575;
  background-color: rgba(247,249,249,.5);
  line-height: 1.3;
}

.c3 {
  color: #1982C4;
}

<header
  className="c0"
>
  <h1
    className="c1"
  >
    Taboo!
  </h1>
  <h3
    className="c2"
  >
    The team game that’s all about what you
     
    <span
      className="c3"
    >
      say,
    </span>
     and what you
     
    <span
      className="c3"
    >
      don’t!
    </span>
  </h3>
</header>
`;

exports[`Header renders correctly on nested home route 1`] = `
.c0 {
  margin-top: 3rem;
  margin-bottom: 3rem;
}

.c1 {
  background: none;
  font-family: 'Oswald',sans-serif;
  font-size: 2.7rem;
  color: #1982C4;
  text-transform: uppercase;
  font-weight: 300;
  font-size: 7rem;
  text-align: center;
  margin: 1rem 0;
}

<header
  className="c0"
>
  <h1
    className="c1"
  >
    Taboo!
  </h1>
</header>
`;

exports[`Header renders correctly when not on home route 1`] = `
.c0 {
  margin-top: 2rem;
  margin-bottom: 3rem;
}

.c1 {
  background: none;
  font-family: 'Oswald',sans-serif;
  font-size: 2.7rem;
  color: #1982C4;
  text-transform: uppercase;
  font-weight: 300;
}

<header
  className="c0"
>
  <h1
    className="c1"
  >
    Taboo!
  </h1>
</header>
`;

exports[`Subheading renders correctly 1`] = `
.c0 {
  font-size: 2.4rem;
  text-align: center;
  width: 30rem;
  margin: 0 auto;
  font-weight: 300;
  color: #757575;
  background-color: rgba(247,249,249,.5);
  line-height: 1.3;
}

<h3
  className="c0"
/>
`;
